src/CommonITILObject.php

```diff
@@ -8607,33 +8607,24 @@ protected function updateActors(bool $disable_notifications = false)
             $updated   = [];
 
             foreach ($actors as $actor) {
-                if (
-                    $actor['items_id'] === 0
-                    && (
-                        ($actor['itemtype'] === User::class && empty($actor['alternative_email'] ?? ''))
-                        || $actor['itemtype'] !== User::class
-                    )
-                ) {
-                    // Empty values, probably provided by static::getDefaultValues()
-                    continue;
-                }
-
                 $found = false;
                 foreach ($existings as $existing) {
-                    if (
-                        $actor['itemtype'] === User::class
-                        && $actor['items_id'] == 0
-                        && $actor['itemtype'] == $existing['itemtype']
-                    ) {
-                        // "email" actor found
-                        if ($actor['alternative_email'] == $existing['alternative_email']) {
-                            $found = true;
-                            break;
+                    if ($actor['itemtype'] === User::class && $actor['items_id'] == 0) {
+                        // This is an anonymous actor
+                        if ($existing['itemtype'] === User::class && $existing['items_id'] == 0) {
```

Comment:
Both if could be combined.

Reply:
Fixed in https://github.com/glpi-project/glpi/pull/16070/commits/436d0b4db1c83391ff6faaa54da6daefd7bda1d7